internal/components/generic_parser.go

```diff
@@ -27,15 +28,48 @@ var (
 	_ Parser = &GenericParser[SingleEndpointConfig]{}
 )
 
+// AddressDefaulter ...
+func AddressDefaulter(logger logr.Logger, defaultRecAddr string, config *SingleEndpointConfig) (*SingleEndpointConfig, error) {
+	if config.Endpoint == "" {
+		return config, nil
+	}
+
+	v := strings.Split(config.Endpoint, ":")
+	if len(v) < 1 {
+		return config, nil
+	}
+	if v[0] == "" {
+		config.Endpoint = fmt.Sprintf("%s:%s", defaultRecAddr, v[1])
+	}
+	return config, nil
+}
+
 // GenericParser serves as scaffolding for custom parsing logic by isolating
 // functionality to idempotent functions.
 type GenericParser[T any] struct {
-	name         string
-	settings     *Settings[T]
-	portParser   PortParser[T]
-	rbacGen      RBACRuleGenerator[T]
-	livenessGen  ProbeGenerator[T]
-	readinessGen ProbeGenerator[T]
+	name            string
+	settings        *Settings[T]
+	portParser      PortParser[T]
+	rbacGen         RBACRuleGenerator[T]
+	livenessGen     ProbeGenerator[T]
+	readinessGen    ProbeGenerator[T]
+	defaultsApplier Defaulter[T]
```

Comment:
youll need to set a functionthat matches the signature here for the single endpoint config.

Reply:
That should be here:
```golang
func NewSinglePortParserBuilder(name string, port int32) Builder[*SingleEndpointConfig] {
	return NewBuilder[*SingleEndpointConfig]().WithPort(port).WithName(name).WithPortParser(ParseSingleEndpoint).WithDefaultsApplier(AddressDefaulter)
}

func NewSilentSinglePortParserBuilder(name string, port int32) Builder[*SingleEndpointConfig] {
	return NewBuilder[*SingleEndpointConfig]().WithPort(port).WithName(name).WithPortParser(ParseSingleEndpointSilent).WithDefaultsApplier(AddressDefaulter)
}
```

Reply:
ah 👍